test/e2e/tests/test_ca_hierarchy.py

```diff
@@ -0,0 +1,241 @@
+    end_entity_secret = create_secret
+
+    replacements = {}
+    replacements["CA_NAME"] = ca_name
+    replacements["CERT_NAME"] = cert_name
+    replacements["CA_ACTIVATION_NAME"] = act_name
+    replacements["SUB_CA_NAME"] = sub_ca_name
+    replacements["SUB_CERT_NAME"] = sub_cert_name
+    replacements["SUB_CA_ACTIVATION_NAME"] = sub_act_name
+    replacements["END_ENTITY_CERT_NAME"] = end_entity_cert_name
+
+    replacements["CERTIFICATE_SEC_NS"] = secret.ns
+    replacements["CERTIFICATE_SEC_NAME"] = secret.name
+    replacements["CERTIFICATE_SEC_KEY"] = secret.key
+    replacements["CERTIFICATE_CHAIN_SEC_NS"] = certificate_chain_secret.ns
+    replacements["CERTIFICATE_CHAIN_SEC_NAME"] = certificate_chain_secret.name
+    replacements["CERTIFICATE_CHAIN_SEC_KEY"] = certificate_chain_secret.key
+
+    replacements["SUB_CERTIFICATE_SEC_NS"] = sub_secret.ns
+    replacements["SUB_CERTIFICATE_SEC_NAME"] = sub_secret.name
+    replacements["SUB_CERTIFICATE_SEC_KEY"] = sub_secret.key
+    replacements["COMPLETE_CERTIFICATE_CHAIN_SEC_NS"] = sub_certificate_chain_secret.ns
+    replacements["COMPLETE_CERTIFICATE_CHAIN_SEC_NAME"] = sub_certificate_chain_secret.name
+    replacements["COMPLETE_CERTIFICATE_CHAIN_SEC_KEY"] = sub_certificate_chain_secret.key
+    
+    replacements["END_CERTIFICATE_SEC_NS"] = end_entity_secret.ns
+    replacements["END_CERTIFICATE_SEC_NAME"] = end_entity_secret.name
+    replacements["END_CERTIFICATE_SEC_KEY"] = end_entity_secret.key
+
+    # Load CRs
+    resource_data = load_all_acmpca_resources(
+        "ca_hierarchy",
+        additional_replacements=replacements,
+    )
+
+    # Create CA reference
+    ca_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificateauthorities",
+        ca_name, namespace="default",
+    )
+
+    # Create Certificate reference
+    cert_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificates",
+        cert_name, namespace="default",
+    )
+    
+    # Create CAActivation reference
+    act_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificateauthorityactivations",
+        act_name, namespace="default",
+    )
+
+    # Create Subordinate CA reference
+    sub_ca_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificateauthorities",
+        sub_ca_name, namespace="default",
+    )
+
+    # Create Subordinate Certificate reference
+    sub_cert_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificates",
+        sub_cert_name, namespace="default",
+    )
+    
+    # Create Subordinate CAActivation reference
+    sub_act_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificateauthorityactivations",
+        sub_act_name, namespace="default",
+    )
+
+    # Create End-entity Certificate reference
+    end_entity_cert_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificates",
+        end_entity_cert_name, namespace="default",
+    )
```

Comment:
If one of these calls somewhat fails, you'll endup with dangling resources, which will increase our testing bill